pkg/ccl/multitenantccl/tenantcostclient/metrics.go

```diff
@@ -175,3 +230,68 @@ func (m *metrics) getConsumption(consumption *kvpb.TenantConsumption) {
 	consumption.CrossRegionNetworkRU = m.TotalCrossRegionNetworkRU.Count()
 	consumption.EstimatedCPUSeconds = m.TotalEstimatedCPUSeconds.Count()
 }
+
+// GetReplicatedBytes returns the replicated bytes metric for the given network
+// path.
+func (m *metrics) GetReplicatedBytes(path tenantcostmodel.LocalityNetworkPath) *aggmetric.Counter {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	// Check if we have cached values.
+	np := networkPath{fromNodeID: path.FromNodeID, toNodeID: path.ToNodeID}
+	if cached, ok := m.mu.cachedPathMetrics[np]; ok {
+		return cached.ReplicatedBytes
+	}
+
+	// Check if we've already initialized the metric. This is the case if a
+	// different network path results in the same normalized locality values.
+	labelValues := makeLocalityLabelValues(m.localLocality, path.FromLocality, path.ToLocality)
+	storeKey := strings.Join(labelValues, ",")
+
+	pm, ok := m.mu.pathMetrics[storeKey]
+	if !ok {
+		pm = networkPathMetrics{
+			ReplicatedBytes: m.TotalReplicatedBytes.AddChild(labelValues...),
+		}
+		m.mu.pathMetrics[storeKey] = pm
+	}
+	m.mu.cachedPathMetrics[np] = pm
+
+	return pm.ReplicatedBytes
+}
+
+// makeLocalityLabelValues returns a list of label values which can be used
+// for network path metrics. This applies a mapping approach where all the
+// values in `from` and `to` are taken up until the first value that differs.
+// Remaining values will be padded as empty strings. Note that the keys must
+// exist in the base locality specified. See test cases for more information.
+func makeLocalityLabelValues(baseLocality, fromLocality, toLocality roachpb.Locality) []string {
+	fromVals, toVals := []string{}, []string{}
+	matching := true
+	i := 0
+	for i < len(baseLocality.Tiers) {
+		if matching {
+			if i >= len(fromLocality.Tiers) ||
```

Comment:
What's the benefit of considering `baseLocality`? Why not just consider `fromLocality` and `toLocality`? What scenario are you thinking about here?